exec/session.go

```diff
@@ -227,13 +228,25 @@ func (s *Session) Must(ctx context.Context, funcv *bigslice.FuncValue, args ...i
 // the task results are needed by another computation, they will be recomputed.
 // Discarding is best-effort, so no error is returned.
 func (s *Session) Discard(ctx context.Context, roots []*Task) {
-	var wg sync.WaitGroup
-	iterTasks(roots, func(task *Task) {
+	var (
+		limiter = limiter.New()
+		wg      sync.WaitGroup
+	)
+	limiter.Release(64)
+	// Best effort, so discard error.
+	_ = iterTasks(roots, func(task *Task) error {
+		if err := limiter.Acquire(ctx, 1); err != nil {
+			return err
+		}
 		wg.Add(1)
 		go func() {
-			defer wg.Done()
+			defer func() {
+				wg.Done()
+				limiter.Release(1)
+			}()
```

Comment:
```suggestion
			defer wg.Done()
			defer limiter.Release(1)
```
(This changes the order of calls, but either order seems fine.)